gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -45,9 +45,13 @@ public EntitiesRequest transformFilter(
     EntitiesRequest.Builder entitiesRequestBuilder = EntitiesRequest.newBuilder(originalRequest);
     // Convert the time range into a filter and set it on the request so that all downstream
     // components needn't treat it specially.
-    Filter filter = TimeRangeFilterUtil.addTimeRangeFilter(
-        context.getTimestampAttributeId(), originalRequest.getFilter(),
-        originalRequest.getStartTimeMillis(), originalRequest.getEndTimeMillis());
+    Filter filter = originalRequest.getFilter();
+    if (originalRequest.getQueryLiveEntitiesOnly()) {
```

Comment:
I'm assuming this check alone is not enough. The TODO you added in ExecutionTreeBuilder is relevant here too right?

Reply:
right. I was thinking to get rid of the check and let it add the time range filter, but then remove the time range filter specifically in the `ExecutionVisitor` based on the flag

So, that we have the check at a single place